src/common/Gateway.sol

```diff
@@ -16,21 +19,36 @@ import {IMessageSender} from "src/common/interfaces/IMessageSender.sol";
 import {IGateway} from "src/common/interfaces/IGateway.sol";
 import {PoolId} from "src/common/types/PoolId.sol";
 import {IGatewayHandler} from "src/common/interfaces/IGatewayHandlers.sol";
+import {MessageLib, MessageType} from "src/common/libraries/MessageLib.sol";
+
+uint8 constant MESSAGE_PROOF_ID = 1;
+
+function deserializeMessageProof(bytes memory data) pure returns (bytes32) {
```

Comment:
Actually this seems unorganized now that I see it.. Maybe we stick to the library

Reply:
> Actually this seems unorganized

In what sense do you say it?

Will refactor this in a future PR, in case we decide to go with a library

Reply:
I think reading the code, it is harder to get lost and not understand the organization of the logic now.

But totally fine to revisit this later.

Reply:
You were right, I'm more happy with: [de87d6f](https://github.com/centrifuge/protocol-v3/pull/268/commits/de87d6fcb54c20eaedbe560d7d57fac3d10fdb90)